tools/autograd/gen_variable_type.py

```diff
@@ -600,6 +677,37 @@ def wrap_view_single(output_var, base_var):
         else:
             return 'as_variable({})'.format(call), []
 
+    def enforce_same_tensorimpl_and_storage(env, call):
+        save_ptrs_block = ''
+        enforce_same_ptrs_block = ''
+        if declaration['name'] not in DONT_ENFORCE_SAME_TENSOR_IMPL_OR_STORAGE:
+            for arg in env.get('unpacked_args', []):
+                simple_type = env['unpacked_args_simple_type'][arg]
+                if simple_type == 'TensorList':
+                    save_ptrs_block += SAVE_STORAGE_AND_IMPL.substitute(
```

Comment:
nit: do you actually have to build these into strings over just keeping them as lists?  I thought CodeTemplates could handle those.

Reply:
Fixed by using lists instead of strings :)